meta-security-smack/recipes-core/systemd/systemd_%.bbappend

```diff
@@ -57,6 +64,62 @@ EXTRA_OECONF_append_smack = " --with-smack-run-label=System"
 # /etc/tmpfiles.d/00-create-volatile.conf
 RDEPENDS_${PN}_append_smack = " smack-userspace"
 do_install_append_smack() {
-    # sed did weird things for this replacement (duplicated ExecStart), works with perl.
-    perl -pi -e "s@^ExecStart=(.*)@ExecStart=\\1\\nExecStartPost=/bin/sh -c '([ ! -d /var/tmp ] || chsmack -L -a \"*\" /var/tmp) && ([ ! -d /var/log ] || chsmack -L -a System::Log /var/log && chsmack -L -t /var/log)'@" ${D}${systemd_unitdir}/system/systemd-tmpfiles-setup.service
+    install_file ${D}${systemd_unitdir}/system/systemd-tmpfiles-setup.service.d/smack.conf <<EOF
+[Service]
+ExecStartPost=/bin/sh -c '([ ! -d /var/tmp ] || chsmack -L -a \"*\" /var/tmp) && ([ ! -d /var/log ] || chsmack -L -a System::Log /var/log && chsmack -L -t /var/log)'
+EOF
+
+    # Mount /tmp publicly accessable. Based on patch by Michael Demeter <michael.demeter@intel.com>.
+    # Upstream systemd temporarily had SmackFileSystemRoot for this (https://github.com/systemd/systemd/pull/1664),
+    # but it was removed again (https://github.com/systemd/systemd/issues/1696) because
+    # util-linux mount will ignore smackfsroot when Smack is not active. However,
+    # busybox is not that intelligent.
+    #
+    # When using busybox mount, adding smackfsroot=* and booting without
+    # Smack (i.e. security=none), tmp.mount will fail with an error about
+    # "Bad mount option smackfsroot".
+    install_file ${D}${systemd_unitdir}/system/tmp.mount.d/smack.conf <<EOF
+[Mount]
+Options=smackfsroot=*
+EOF
+
+    # Run systemd-journald with the hat ("^") Smack label.
+    #
+    # The journal daemon needs global read access to gather information
+    # about the services spawned by systemd. The hat label is intended
+    # for this purpose. The journal daemon is the only part of the
+    # System domain that needs read access to the User domain. Giving
+    # the journal daemon the hat label means that we can remove the
+    # System domain's read access to the User domain.
+    #
+    # Original author: Casey Schaufler <casey@schaufler-ca.com>
+    #
+    # This is considered a configuration change and thus distro specific.
+    install_file ${D}${systemd_unitdir}/system/systemd-journald.service.d/smack.conf <<EOF
```

Comment:
On Wed, 2016-01-13 at 04:51 -0800, Ismo Puustinen wrote:
> In meta-security-smack/recipes-core/systemd/systemd_%.bbappend:
> 
> > +Options=smackfsroot=*
> > +EOF
> > +
> > +    # Run systemd-journald with the hat ("^") Smack label.
> > +    #
> > +    # The journal daemon needs global read access to gather information
> > +    # about the services spawned by systemd. The hat label is intended
> > +    # for this purpose. The journal daemon is the only part of the
> > +    # System domain that needs read access to the User domain. Giving
> > +    # the journal daemon the hat label means that we can remove the
> > +    # System domain's read access to the User domain.
> > +    #
> > +    # Original author: Casey Schaufler <casey@schaufler-ca.com>
> > +    #
> > +    # This is considered a configuration change and thus distro specific.
> > +    install_file ${D}${systemd_unitdir}/system/systemd-journald.service.d/smack.conf <<EOF
> 
> Do you need to do it like this because of chmod?

Yes.

>  Or do you modify a file that you expect to be there (because of the
> >> instead of > in the install_file function)?

That's unintentionally, > would be sufficient. The file is expected to
not exist yet.

>  In that case you can just add another configuration file and have it
> appended to the .service file.

I found it easier to see what goes on when embedding the config
fragments inside the .bbappend. Having three files each with a few lines
seemed overkill to me. But that's arguably a bit subjective.

You mean, append at the end of the existing tmp.mount instead of
creating another file in tmp.mount.d?

I'm not sure whether it is legal to have the same section multiple times
in a unit file. The .d mechanism seems cleaner.